kore/src/Kore/Builtin/Bool.hs

```diff
@@ -163,48 +165,69 @@ builtinFunctions =
     xor a b = (a && not b) || (not a && b)
     implies a b = not a || b
 
--- | Unification of @BOOL.Bool@ values.
+data UnifyBool = UnifyBool
+    { bool1, bool2 :: !InternalBool
+    }
+
+-- | Matches two Bool values.
+matchBools ::
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    Maybe UnifyBool
+matchBools first second
+    | InternalBool_ bool1 <- first
+      , InternalBool_ bool2 <- second =
+        Just UnifyBool{bool1, bool2}
+    | otherwise = Nothing
+{-# INLINE matchBools #-}
+
+-- | When bool values are equal, returns first term; otherwise returns bottom.
 unifyBool ::
-    forall variable unifier.
-    InternalVariable variable =>
+    forall unifier.
     MonadUnify unifier =>
-    TermLike variable ->
-    TermLike variable ->
-    MaybeT unifier (Pattern variable)
-unifyBool a b =
-    worker a b <|> worker b a
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
```

Comment:
We should not pass in the original `TermLike`s to any `unify...` function.

Reply:
We are putting this off for the time being per our discussion at the standup about the `explainAndReturnBottom` function.